src/include/migraphx/op/dequantizelinear.hpp

```diff
@@ -49,10 +49,13 @@ struct dequantizelinear
     std::string name() const { return "dequantizelinear"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.same_dims().has(2, 3);
-        if(inputs.size() == 3 and inputs[0].type() != inputs[2].type())
+        if(inputs.size() == 3)
         {
-            MIGRAPHX_THROW("DEQUANTIZELINEAR: Zero point and input should be the same type.");
+            check_shapes{inputs, *this}.same_dims().has(2, 3);
```

Comment:
Why is this moved down here? We should still check that the inputs are the same dims even when 2 inputs are given. It makes sense up at the top.

Reply:
That wont be the case though in this operator would it? I guess we broadcast out scale if its a scalar or its 1-d for the case of scale being a tensor

Reply:
That's where the issue was coming from when invoking dequantize_linear, zero points weren't present and resulted in an error.

Reply:
What was the error specifically? This allows either 2 or 3 inputs so it shouldn't throw an error for 2 inputs.

Reply:
![image](https://github.com/user-attachments/assets/1f320add-9f9e-4a2d-a28e-2686d6540825)

would invoke

```[   RUN    ] matmulintegertofloat_test
terminate called after throwing an instance of 'migraphx::version_1::exception'
  what():  /code/AMDMIGraphX/src/include/migraphx/check_shapes.hpp:220: same_dims: dequantizelinear: Dimensions do not match```

Reply:
Unless you want us to always broadcast the scale dimensions (even if its 1-d) but then that means we're inserting a broadcast to remove later. For the scale here, we're never supposed to have the same dimensions between input and scale vectors. Its why I gated this check by input size.